packages/api/app/Controllers/Http/TechnologyController.js

```diff
@@ -52,9 +83,15 @@ class TechnologyController {
 
 		const data = getFields(request);
 
+		if (data.termId) {
```

Comment:
Confesso que achei essa parte implícita aqui foi a única parte que achei um pouco estranha. Mas tá um pouco estranho: se eu passo um `termId` significa que eu quero atualizar só o `termId` da tecnologia. Se não, eu atualizo a technology em si. 

Eu acho que faz mais sentido **OU** utilizar o padrão mesmo, que é inclusive o seguindo pelo Adonis:
Para atualizar a tecnologia:
``` 
technology/:id
```
Para atualizar o `termId` de uma tecnology
```
technology/:id/termId
```
**OU**
deixar de um modo que o usuário possa **atualizar ambos de uma vez**. Ou seja, se ele passar um novo `termId`, você atualiza **tudo, inclusive o termId**. Se ele não passar, você atualiza a tecnologia em si. O que acha?

Eu gosto mais da 2ª ideia, porque não precisamos alterar a rota e, mesmo assim, permitimos alterar ambos de uma vez só.

Reply:
@lcnogueira mas a forma com que Alexandre implementou foi seguindo essa segunda ideia não?

A ideia é que baseado na payload (e não no endpoint) o controller atualizar a tecnologia e seus termos associados.

Reply:
Acho que não, @nicholasio . Como tem um `return` dentro do if e o merge dos dados da tecnologia só acontece depois do if, eu acho que aqui **ou** atualiza os termos **ou** atualiza a tecnologia. Eu não consigo atualizar, por exemplo, likes e termos. Acho que ficou assim. Confere, @alexandreadames ?

Reply:
Exato @lcnogueira, eu vou modificar segundo o que você falou (ambos de uma só vez), eu fiz isso apenas porque no caso de incluir um termo em uma tecnologia o retorno é um, e se for a atualização dos dados da tecnologia o retorno é outro. Vou tentar unificar em um só retorno.

Reply:
Mas se você ou @nicholasio enxergar algo que vai ficar mais fácil deixando assim, pode deixar viu, @alexandreadames ? Eu só comentei aqui porque, em um primeiro momento, faz mais sentido retornar na mesma rota os dados sempre no mesmo formato, a meu ver. Ex.: pode retornar sempre a tecnologia com os itens associados, por exemplo...independentemente se atualizou algum outro campo ou algum item. **Mas se tiver algo em mente, que já vai ser facilitado por isso aí, deixa**. Sem pro :+1: .